java/java.lsp.server/vscode/src/panels/SshGuidePanel.ts

```diff
@@ -0,0 +1,132 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as vscode from "vscode";
+import * as path from "path";
+import * as fs from "fs";
+import * as Handlebars from "handlebars";
+import { promisify } from "util";
+import { toggleGuideFor, shouldHideGuideFor, removeGuidePropertiesFromStateFor, NotificationActionType, sshConfigLocation, StatePropertyPrefix, dummyKeyPathLocation } from "./guidesUtil";
+
+const readFile = promisify(fs.readFile);
+
+export class SshGuidePanel {
+    public static currentPanel: SshGuidePanel | undefined;
+    public static readonly viewType: string = "sshGuide";
+
+    private static readonly actionType: NotificationActionType = 'ssh-connection';
+    private static readonly statePropertyPrefix: StatePropertyPrefix = 'sshGuide';
+
+    private static readonly webviewsFolder: string = "webviews";
+    private static readonly scriptsFolder: string = "scripts";
+    private static readonly stylesFolder: string = "styles";
+
+    private readonly panel: vscode.WebviewPanel;
+    private disposables: vscode.Disposable[] = [];
+
+    public static createOrShow(context: vscode.ExtensionContext) {
+        if (SshGuidePanel.currentPanel) {
+            SshGuidePanel.currentPanel.panel.reveal();
+            return;
+        }
+
+        SshGuidePanel.currentPanel = new SshGuidePanel(context);
+    }
+
+    private constructor(context: vscode.ExtensionContext) {
+        this.panel = vscode.window.createWebviewPanel(
+            SshGuidePanel.viewType,
+            "Oracle Cloud Assets",
+            { viewColumn: vscode.ViewColumn.One, preserveFocus: true },
+            {
+                enableCommandUris: true,
+                enableScripts: true,
+                localResourceRoots: [vscode.Uri.file(path.join(context.extensionPath, SshGuidePanel.webviewsFolder))],
+            }
+        );
+        const iconPath: vscode.Uri = vscode.Uri.file(path.join(context.extensionPath, "images", "Apache_NetBeans_Logo.png"));
+        this.panel.iconPath = {
+            light: iconPath,
+            dark: iconPath,
+        };
+
+        this.setHtml(context);
+
+        this.panel.onDidDispose(
+            () => {
+                this.dispose();
+                removeGuidePropertiesFromStateFor(context, SshGuidePanel.statePropertyPrefix);
+            },
+            null,
+            this.disposables
+        );
+
+        this.panel.onDidChangeViewState(
+            () => {
+                if (this.panel.visible) {
+                    this.setHtml(context);
+                }
+            },
+            null,
+            this.disposables
+        );
+
+        this.panel.webview.onDidReceiveMessage(
+            (message) => {
+                if (message.command === "showSSHGuide") {
```

Comment:
note: if doing an abstract class, the message command can be just one for all guides: the view will be talking only to its own panel instance, right ?